src/gorilla.c

```diff
@@ -0,0 +1,261 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include <assert.h>
+
+#include "gorilla.h"
+
+
+#define _64BIT 64
+#define DOUBLE_LEADING 5
+#define DOUBLE_BLOCK_SIZE 6
+#define DOUBLE_BLOCK_ADJUST 1
+
+#define CHECKSPACE(chunk, x)    if (!isSpaceAvailable(chunk, x)) { \
+                                    return CC_ERR; }
+
+#define  LeadingZeros64(x)  __builtin_clzll(x)
+#define TrailingZeros64(x)  __builtin_ctzll(x)
+
+// Clear all bits from position `bits`
+static u_int64_t clearBits(u_int64_t data, u_int64_t bits) {
+    return data & ((1LL << bits) - 1);
+}
+
+// Converts `x`, an int64, to binary representation with length `l`
+static u_int64_t int2bin(int64_t x, u_int8_t l) {
+    u_int64_t negative = x < 0 ? 1 : 0;
+    u_int64_t rv = clearBits(x, l - 1);
+    rv |= (negative << (l - 1));
+    return rv;
+}
+
+// Converts `x`, a binary of length `l`, into an int64
+static int64_t bin2int(u_int64_t x, u_int8_t l) {
+    bool negative = x & (1LL << (l - 1));
+    int64_t rv = x;
+    if (negative) {
+        rv -= 1LL << l;
+    }
+    return rv;
+}
+
+// Return `true` if byte is off
+static bool isBitOff(u_int64_t *x, u_int64_t idx) {
```

Comment:
Its worth calling this isFirstBitOff

Reply:
It isn't just for the first bit but for any bit